usage.go

```diff
@@ -218,7 +218,15 @@ func (p *Parser) writeHelpForSubcommand(w io.Writer, cmd *command) {
 		}
 	}
 
-	if p.description != "" {
+	var description string
+	if v := p.val(cmd.dest); v.IsValid() && !isZero(v) {
```

Comment:
I don't think it makes sense to include `!isZero(v)` in here because `v.IsValid` already checks that (see https://cs.opensource.google/go/go/+/refs/tags/go1.17.5:src/reflect/value.go;l=1435)

The basic issue with this approach is that it won't work if the struct for this subcommand has a parent that is nil, since in that case we will end up [here](https://github.com/alexflint/go-arg/blob/cf2205c84db73b0c2be62a8981c2f61399b32d4a/parse.go#L685) and then `v.IsValid()` will return true.

It would be better to store the type for each subcommand in the `command` struct and then instantiate a new struct of that type right here and check whether it satisfies the `Described` interface